compiler/rustc_metadata/src/locator.rs

```diff
@@ -549,6 +552,13 @@ impl<'a> CrateLocator<'a> {
                             continue;
                         }
                     }
+                    Err(MetadataError::LoadFailure(err)) => {
+                        // The file was present and created by the same compiler version, but we
+                        // couldn't load it for some reason.  Give a hard error instead of silently
+                        // ignoring it, but only if we would have given an error anyway.
+                        self.rejected_via_invalid.push(CrateMismatch { path: lib, got: err });
+                        continue;
+                    }
                     Err(err) => {
```

Comment:
It would be better to make this matching exhaustive to see which errors we are skipping silently.